release.sh

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+
+set -e
+
+current_branch=$(git branch --show-current)
+git_changes=$(git status -s | wc -l)
+if [[ $git_changes != 0 ]]; then
+	if [[ $current_branch == "master" ]]; then
+		echo "You have modified the master branch."
+		echo "Please revert or move your changes to another branch before running this script."
+	else
+		echo "Please commit or stash your changes before running this script"
+	fi
+	exit 1
+fi
+
+git fetch
+ahead=$(git status -sb | grep ahead -c)
+if [[ $ahead != 0 ]]; then
+	echo "Your local master branch is ahead of the remote master branch. Exiting."
+	exit 1
+fi
+
+
+release_type=$1
+if [[ $release_type != "minor" && $release_type != "patch" && $release_type != "major" ]]; then
+	echo "Invalid argument: $release_type"
+	exit 1
+fi
+
+read -ep "Are you sure you want to cut a $release_type release? [y/N] " continue
+if [[ $continue != "yes" && $continue != "y" && $continue != "Yes" && $continue != "YES" ]]; then
+	exit
+fi
+
+git checkout master
+git pull origin master
+
+version=v$(cat package.json| grep -Po -m 1 '\d+\.\d+\.\d+')
+npm version $release_type --no-git-tag-version
+new_version=v$(cat package.json | grep -Po -m 1 '\d+\.\d+\.\d+')
+
+echo "Cutting $release_type release"
+echo "$version -> $new_version"
+
+git add .
+git commit -m "build(release): Release $new_version"
+git tag -a $new_version HEAD -m "Release $new_verion"
```

Comment:
If the PR merge strategy is "Squash and merge", we'll lose the commit this tag connects to. I wonder if we really even need a PR at all. If the script is the only thing doing work, and all we're doing is updating `package.json`, maybe a direct commit to master would suffice. I think we also want to used gpg signing for our release tags.

Reply:
The intent for this is to be a direct commit to master. I'm unfamiliar with GPG signing for git tags, but if it helps make things more secure, I'm for it.

Reply:
If we do want to push directly to master, and it's *only* a change to `package.json`, a direct `git push` should be fine, i.e. no PR needed. I'd add a very strongly worded comment at the top of this script (I like to use words like `ACHTUNG` and `PLEASE FOR THE LOVE OF PETE READ ME BEFORE MESSING WITH THIS FILE`) that points out that this script should change the _minimum_ amount of things, in the most mechanical way possible. Might even be good to run some sort of wellness check after the change to `package.json` to make sure that something didn't get corrupted and you just released a version with a typo that prevents it from working (not the end of the world, but maybe a little embarrassing).